### PR TITLE
added options to CRF model metadata

### PR DESCRIPTION
**Pull request by behrica**

### CODE REVIEW

File: deps.edn [hunk 1]
```diff
@@ -3,7 +3,7 @@
            aysylu/loom                       {:mvn/version "1.0.2"}
            camel-snake-kebab                 {:mvn/version "0.4.2"}
            edu.stanford.nlp/stanford-corenlp {:mvn/version "4.2.0"}
-           techascent/tech.ml                {:mvn/version "5.01"}
+           scicloj/metamorph.ml              {:mvn/version "0.3.0-alpha3"}
```
Comment (behrica) [inline deps.edn:6]: There is a transition from `tech.ml` ongoing to:
https://github.com/scicloj/metamorph.ml - branch mergeTechML 
https://github.com/scicloj/scicloj.ml  (which is a shim without code, just re-organizing namespaces from various projects)

The "feature" of documenting options as part of model metadata is not in `tech.ml`, and will likely not go there ever.
If all goes well, `tech.ml` will be feature frozen and deprecated soon. 

So we need it "successor" here , namely `methamorph.ml`

File: test/dk/simongray/test/ml_test.clj [hunk 2]
```diff
@@ -2,15 +2,26 @@
...
+  (let [model      (ml/train train-ds {:model-type :corenlp/crf})
         text-ds    (ds/->dataset {:column-0 ["I like Mr. X and I like  Mr. Y" "Smith" "Mrs." "Weston"]})
         prediction (ml/predict text-ds model)]
     (is (= [["PERS" 7 10]
             ["PERS" 25 30]]
            (first (:ner prediction))))))
+
+(deftest train-predict-defaul-no-ngrams
+  (let [model      (ml/train train-ds {:model-type :corenlp/crf
+                                      :useNGrams "false"})
```
Comment (behrica) [inline test/dk/simongray/test/ml_test.clj:22]: Model options would be specified like this

File: deps.edn (hunk 1 above)
Comment (simongray) [reply deps.edn:6]: My apologies, but I don't think I quite get the relations between the different projects... metamorph.ml is succeeding tech.ml, but the idea is to merge the code into scicloj.ml in the future, so metamorph.ml is just another step on the way towards a scicloj.ml? But why the sudden deprecation of tech.ml?

File: test/dk/simongray/test/ml_test.clj (hunk 2 above)
Comment (simongray) [reply test/dk/simongray/test/ml_test.clj:22]: Yup, let's go with that for now. The more I think about it, the more I think I prefer keeping the default case from CoreNLP. Ideally people should be able to copy-paste names they find in CoreNLP documentation.

File: src/dk/simongray/datalinguist/ml/crf.clj [hunk 3]
```diff
@@ -101,4 +100,19 @@
...
 
-(ml/define-model! :corenlp/crf train predict {})
+
+
+(def metamorph-ml-options
+  (map
+   (fn [[k v]]
+     (hash-map :name k :type :string :default v))
+   default-opts
+   ))
```
Comment (simongray) [inline src/dk/simongray/datalinguist/ml/crf.clj:110]: What's with the format of the options here? I would expect a map, but this is a sequence of maps?

```clojure
({:default "true", :name :useClassFeature, :type :string}
 {:default "true", :name :useTypeSeqs, :type :string}
 {:default "true", :name :usePrevSequences, :type :string}
 {:default "true", :name :useDisjunctive, :type :string}
 {:default "true", :name :useWord, :type :string}
 {:default "ner-model.ser.gz", :name :serializeTo, :type :string}
 {:default "true", :name :useNext, :type :string}
 {:default "true", :name :useSequences, :type :string}
 {:default "true", :name :useTypeySequences, :type :string}
 {:default "true", :name :usePrev, :type :string}
 {:default "1", :name :maxLeft, :type :string}
 {:default "true", :name :useTypeSeqs2, :type :string}
 {:default "none", :name :wordShape, :type :string}
 {:default "true", :name :noMidNGrams, :type :string}
 {:default "true", :name :useNGrams, :type :string}
 {:default "6", :name :maxNGramLeng, :type :string})
```

**Review by simongray — APPROVED** [review]

**Pull request merged**